Source/WTF/wtf/text/StringCommon.h

```diff
@@ -767,8 +849,22 @@ ALWAYS_INLINE static size_t reverseFindInner(std::span<const SearchCharacterType
     return delta;
 }
 
-template<typename CharacterType> inline bool equalLettersIgnoringASCIICaseWithLength(std::span<const CharacterType> characters, std::span<const Latin1Character> lowercaseLetters, size_t length)
+template <typename SearchCharacterType, typename MatchCharacterType> requires(EqualityComparable<SearchCharacterType, MatchCharacterType>)
+ALWAYS_INLINE static size_t reverseFind(std::span<const SearchCharacterType> searchCharacters, std::span<const MatchCharacterType> matchCharacters, size_t start = std::numeric_limits<size_t>::max())
+{
+    return reverseFindInner(searchCharacters, matchCharacters, start);
+}
+
+template <typename CharacterType> requires(OneByteCharacterType<CharacterType>)
+ALWAYS_INLINE static size_t reverseFind(std::span<const CharacterType> searchCharacters, ASCIILiteral matchCharacters)
 {
+    return reverseFind(searchCharacters, byteCast<CharacterType>(matchCharacters.span()));
+}
+
+template<typename CharacterType, typename OneByteCharacterType> requires(std::is_same_v<std::remove_const_t<CharacterType>, std::remove_const_t<OneByteCharacterType>> || std::is_same_v<std::remove_const_t<OneByteCharacterType>, Latin1Character>)
```

Comment:
We should have a concept that is the two String character types rather than writing it out like this.

```suggestion
template<typename CharacterType, typename OneByteCharacterType>
    requires(std::same_as<std::remove_const_t<CharacterType>, std::remove_const_t<OneByteCharacterType>> || std::same_as<std::remove_const_t<OneByteCharacterType>, Latin1Character>)
```